src/exportImage.py

```diff
@@ -121,49 +147,92 @@ def delete_local_directory(self, directoryName):
         self.log.info(
             f'Folder "{directoryName}" in path: "{directory_path}" removed successfully')
 
-    def create_geopackage(self, bbox, filename, url, taskid, fullPath, resolution):
+    def create_geopackage(self, bbox, filename, url, task_id, full_path, max_zoom, resolution):
+        warp_result = self.create_geopackage_base(
+            bbox, filename, url, task_id, full_path, resolution)
+
+        if not warp_result:
+            self.log.error(f'gdal return empty response for task: "{task_id}"')
+            return False
+
+        overviews_result = self.create_geopackage_overview(
+            filename, task_id, full_path, max_zoom)
+
+        # Clear cache (for closing process)
+        warp_result.FlushCache()
+        warp_result = None
+
+        # Check for overview build errors
+        if overviews_result == GDALBuildOverviewsResponse.CE_Failure or overviews_result == GDALBuildOverviewsResponse.CE_Fatal:
+            self.log.error(
+                f'gdal return empty response for task: "{task_id}" (overviews)')
+            return False
+
+        return True
+
+    def create_geopackage_base(self, bbox, filename, url, task_id, full_path, resolution):
         output_format = self.__config["gdal"]["output_format"]
-        es_obj = {"taskId": taskid, "filename": filename}
-        self.log.info(f'Task Id "{taskid}" in progress.')
+        es_obj = {"taskId": task_id, "filename": filename}
+        self.log.info(f'Task Id "{task_id}" in progress.')
         thread_count = self.__config['gdal']['thread_count'] if int(self.__config['gdal']['thread_count']) > 0 \
             else 'val/ALL_CPUS'
         thread_count = f'NUM_THREADS={thread_count}'
         kwargs = {
             'dstSRS': self.__config['gdal']['output_srs'],
             'format': output_format,
             'outputBounds': bbox,
-            'callback': self.progress_callback,
+            'callback': self.warp_progress_callback,
             'callback_data': es_obj,
             'xRes': resolution,
             'yRes': resolution,
             'creationOptions': ['TILING_SCHEME=InspireCrs84Quad'],
             'multithread': self.__config['gdal']['multithread'],
             'warpOptions': [thread_count]
         }
-        result = gdal.Warp(fullPath, url, **kwargs)
-        if result:
-            result.FlushCache()
-            result = None
-            return True
-        else:
-            self.log.error(f'gdal return empty response for task: "{taskid}"')
-            return False
+        result = gdal.Warp(full_path, url, **kwargs)
+        self.log.info(f'Task Id "{task_id}" completed.')
```

Comment:
it's not completed as the log said - only first stage is completed (geopackage creation)

Reply:
Moved and corrected log